### ISSUE TITLE
Image behavior 

### ISSUE THREAD

**Issue by gregBerthelot** [jpuri/react-draft-wysiwyg#227]
Hi

Currently adding image between 2 words split content into 2 <p> which suppose img is considered as a block element but according to https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements img is an inline element. 

`<p>word1 word2</p>`

Insert img between the 2 words result to
```
<p>word1</p>
<img src="somePath">
<p>word2</p>
```
http://g.recordit.co/sDbYcekhMv.gif

We should get

`<p>word1 <img src="somePath"> word2</p>`

http://g.recordit.co/3cxT6TBghM.gif

This is the only behavior that allow user to insert image at the cursor position, then choose the alignment he want to apply:
- none (stay at the insertion point)
- left: float left
- right: float right

Centering image could be an option to make image as a block element and make it centered in the RTE but in fact as img is an inline element the center option is not necessary, using text-align: center should do the trick.

**Comment by jpuri** [comment]
Hey @gregBerthelot: I see your point, I will look into it.

**Comment by gregBerthelot** [comment]
@jpuri did you get time to look into this enhancement?

**Comment by liSong5713** [comment]
I'm also looking forward to.please .

**Comment by gregBerthelot** [comment]
@jpuri any chance you get time to look into it?

**Comment by jpuri** [comment]
Hi @gregBerthelot , not a lot of time, but I will soon  try to look at this.

**Comment by gregBerthelot** [comment]
Hi Jyoti,

I think image feature should be also enhanced with the ability to change image size once added to the editor.

Cheers
Greg

**Comment by nicolastakashi** [comment]
Any update about this? It's very useful

**Comment by jpuri** [comment]
Sorry no updates, I did not got time for this one yet.

**Comment by gorjan-mishevski** [comment]
Can you at least please point us to to implement this ourselves? This is a deal breaker for me

**Comment by jpuri** [comment]
Currently image size can only be set when image is added, resize feature can be added by allowing user to change size either using mouse or an input. Image block properties can then be update to new size. Hope this helps @gorjan-mishevski .

**Comment by gorjan-mishevski** [comment]
@jpuri Thanks, yes, every bit help. I am still hacking around, do you know of a clean way I can extend it with a draft plugin. I need a advanced version for editing images (Opening a file browser for the user to choose from storage). Or am I boxed in? I can do a PR if I can make this thing work.

**Comment by jpuri** [comment]
Opening file browser already works, check this: https://github.com/jpuri/react-draft-wysiwyg/blob/master/stories/ImageUpload/index.js.

**Comment by gorjan-mishevski** [comment]
Oh cool. Thank you very much! You are a star 🌟 One last question I promise I wont bother. I am new to this draft.js still learning a lot. Is it possible to extend this awesome wysiwyg editor. Becouse its build upon draft I guess I can somehow add new features to it. Lets say I want to implement what the frist guy who created this issue wanted. Can I create a plugin that is easy to extend. Thank you

**Comment by jpuri** [comment]
Editor is extensible in way that it allows adding new toolbar options. You can use it as controlled component and do any changes to editor state. Custom block rendering functions can also be written.

Here is example of custom toolbar options: https://github.com/jpuri/react-draft-wysiwyg/blob/master/stories/CustomToolbar/index.js#L10

But draft has a learning curve :(

I hope that helps, let me know if you have more queries.

**Comment by gorjan-mishevski** [comment]
Yeah, draft is hard. Facebook really likes to complicate things, I love it :)
No more queries for me, thank you for your time. Have a great day ahead.

**Comment by revskill10** [comment]
I need this feature (resizing image after being uploaded). Do you have any roadmap for this feature ?

**Comment by calvinkei** [comment]
Any update on this?

**Comment by hodgef** [comment]
I would recommend anyone in this situation to use [Jodit](https://xdsoft.net/jodit/examples/intergration/react-jodit.html) instead. It is very easy to implement and the editing is very good out of the box.

Sadly react-draft-wysiwyg has been a painful experience for me. Hours upon hours trying to make things work. I hope to spare someone the hassle.
